ddl/ddl_api.go

```diff
@@ -3926,6 +3926,9 @@ func checkModifyTypes(ctx sessionctx.Context, origin *types.FieldType, to *types
 	}
 
 	err = checkModifyCharsetAndCollation(to.Charset, to.Collate, origin.Charset, origin.Collate, needRewriteCollationData)
+	if err != nil && to.Charset == charset.CharsetGBK || origin.Charset == charset.CharsetGBK {
+		return err
+	}
 	// column type change can handle the charset change between these two types in the process of the reorg.
 	if err != nil && errUnsupportedModifyCharset.Equal(err) && canReorg {
 		return nil
```

Comment:
Could we update this code like ```
```suggestion
	if err != nil {
		if to.Charset == charset.CharsetGBK || origin.Charset == charset.CharsetGBK{
			errors.Trace(err)
		}
		// Column type change can handle the charset change between these two types in the process of the reorg.
		if errUnsupportedModifyCharset.Equal(err) && canReorg {
			return nil
		}
	} 
```
```